proxy/proxy.go

```diff
@@ -209,14 +203,24 @@ func InitPassiveHealthChecker(o map[string]string) (bool, *PassiveHealthCheck, e
 				return false, nil, fmt.Errorf("passive health check: invalid maxDropProbability value: %s", value)
 			}
 			result.MaxDropProbability = maxDropProbability
+			requiredKeysInitialized[key] = struct{}{}
+
+		/* optional parameters */
+		case "min-drop-probability":
+			minDropProbability, err := strconv.ParseFloat(value, 64)
+			if err != nil {
+				return false, nil, fmt.Errorf("passive health check: invalid minDropProbability value: %s", value)
+			}
+			if minDropProbability < 0 || minDropProbability > 1 {
+				return false, nil, fmt.Errorf("passive health check: invalid minDropProbability value: %s", value)
+			}
+			result.MinDropProbability = minDropProbability
 		default:
 			return false, nil, fmt.Errorf("passive health check: invalid parameter: key=%s,value=%s", key, value)
 		}
-
-		keysInitialized[key] = struct{}{}
 	}
 
-	if len(keysInitialized) != 4 {
+	if len(requiredKeysInitialized) != 3 {
```

Comment:
If we only use length then integer counter would be enough but I think we can also show which required are missing in the error message, see snippet.